components/timezone/src/error.rs

```diff
@@ -0,0 +1,21 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use displaydoc::Display;
+
+#[cfg(feature = "std")]
+impl std::error::Error for TimeZoneError {}
+
+/// A list of possible error outcomes for working with types in this crate
+/// and operations.
+#[derive(Display, Debug, Copy, Clone, PartialEq)]
+#[non_exhaustive]
+pub enum TimeZoneError {
```

Comment:
Question: Why did you make the errors less expressive?

Reply:
Because the DateTime error exists for multiple different fields whereas we only have one field here